x/bank/migrations/v3/store.go

```diff
@@ -81,7 +81,7 @@ func migrateDenomMetadata(store sdk.KVStore, logger log.Logger) error {
 
 	for ; oldDenomMetaDataIter.Valid(); oldDenomMetaDataIter.Next() {
 		oldKey := oldDenomMetaDataIter.Key()
-		l := len(oldKey)/2 + 1
+		l := len(oldKey) / 2
```

Comment:
Can we add a changelog?

Reply:
I will add a changelog in the 0.46 branch then, so that I don't need to remove it from main later :)

Reply:
Given that this code had a bug, can we add some code comments around whats going on here?

Reply:
@ValarDragon `v2.DenomMetadataPrefix` byte is already cut out by the prefix store -> see line 77 (few lines above), so shouldn't be counted here.